### PR TITLE
Testing the TIC-TAC-TOE game with Rspec

### PR DESCRIPTION
**Pull request by charlyeneh**
This project is the testing of the TIC-TAC-TOE game with Rspec.
We installed Rspec and passed in the necessary tests to our TIC-TAC-TOE game code.

### CODE REVIEW

**Review by smcommits — CHANGES_REQUESTED** [review]
# Status: Changes Required
## Hello, team! You have done a great job. I can see that you have written test cases for all the methods in your application. Although you have done a remarkable job, some changes are still required to improve it further. 

## Required Changes
- [ ] **Specs should cover positive and negative scenarios**

commonsense 
Screenshot from Pathwright. 

Your tests should always cover both, the positive and the negative scenario of something. For instance, if something should return True for some test case, you should also check if it returns False for the opposite test case. 

For example, suppose there is a spec for correct usage of the method [e.g. it "creates object if all required attributes are passed"]
and also suppose there is the spec for the wrong usage of a method [e.g. it "does not create the object if required attributes are missing"].

_Please try to analyze what part of your application can have opposite test cases as well_

- [ ] **In your game_spec.rb, in the valid_move describe block, you are testing that "it 'should return false if a number is not between 0 and 9" but in your expect block you are expecting the return value to be true. Please fix this.**

# 
### Please don't hesitate to contact me on [Slack](https://microverse-students.slack.com/team/U017N7LLZ40) if you need assistance or believe I have overlooked something and made an unfair request.

**Comment by promise-J** [comment]
Added the negative scenerio in both the valid move method and the game ended method as instructed by the previous TSE.

File: spec/game_spec.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
+require './lib/game'
+require './lib/player'
+describe Game do
+  game_class = Game.new
+  game_class.player1 = Player.new('promise', 'o')
+  game_class.player2 = Player.new('alicia', 'x')
+
+  describe '#valid_move' do
+    it 'should return false if a number is not between 1 and 9.' do
+      # game_class.board.board_cell = ['x', 'o', 'x', 'o', 'x', 'o', 'x', 'o', 'x']
```
Comment (cyonii) [inline spec/game_spec.rb]: Do not comment pieces of code

File: spec/game_spec.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    it 'should return true if a move is valid.' do
+      expect(game_class.valid_move(8)).to be true
+    end
+    it 'should return false if a move is invalid.' do
+      expect(game_class.valid_move(10)).to be false
+    end
+  end
+  describe '#game_ended' do
+    it 'should return true if the play count is 9. (a draw)' do
+      #   game_class.board.board_cell = %w[x o x o x o x o x]
```
Comment (cyonii) [inline spec/game_spec.rb]: Do not comment pieces of code

**Review by cyonii — CHANGES_REQUESTED** [review]
# Status: Changes Required ♻️✍️

Hello team, you have done a great job implementing the requested changes by the previous Reviewer, but there are still a few minor defects in your code that you have to improve.

## To Improve
- [ ] Do NOT comment fragments of code, if you need to review a past version of your code, use git history. [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))
- [ ] Your game allows empty input for player name and icon, please fix this so that it keeps prompting the user for input until they provide a valid input.
- [ ] It also allows player one and two to play with the same icons, player's icons should be unique.

Keep up the great effort. Happy coding, and stay motivated. 💻🤓

### You can find me on slack 👉 [CY Kalu](https://microverse-students.slack.com/team/U01494U96FR) if you need further assistance or clarification. I'm always available and will be glad to help.

**Comment by promise-J** [comment]
Added the functionality in the validation of empty username or icon for the tic tac toe game. Also removed the inline comments.

**Review by GzimAsani — APPROVED** [review]
# STATUS: Project Accepted :green_circle: :muscle: 

### Hello team, good job implementing the required changes asked from previous reviewers :+1: , you have met all the requirements and you can merge this pull request.

-[ OPTIONAL ] Please fix your lines in the table it is not looking very good. 
Screenshot from 2021-03-20 01-23-44

**Comment by promise-J** [comment]
Project rspec tic tac toe approved.

**Pull request merged**